cmd/syncthing/gui.go

```diff
@@ -236,14 +248,17 @@ func (s *apiService) Serve() {
 			// We let this be a loud user-visible warning as it may be the only
 			// indication they get that the GUI won't be available.
 			l.Warnln("Starting API/GUI:", err)
-			return
+
+		case <-s.firstStartupFailed:
```

Comment:
How does this work? Why can't startedOnce handle this?

Reply:
`startedOnce` only gets close after a successful start including all setup and logging. I am not sure whether that's relevant or if we could just close it at the beginning of `Serve`, in which case `firstStartupFailed` is indeed redundant. I just went with the smallest change possible.  
Having a closer look I'd say all of the mux setup should probably be in `newApiService` anyway.